.github/workflows/macos.yml

```diff
@@ -61,36 +61,36 @@ jobs:
         restore-keys: |
           ${{ github.workflow }}-${{ matrix.python-version }}-ccache-macos-
 
+    - name: Setup Conda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        mamba-version: "*"
+        channels: conda-forge
+        channel-priority: true
+        activate-environment: scipy-dev
+        use-only-tar-bz2: true
+
     - name: Cache conda
       uses: actions/cache@v2
       env:
         # Increase this value to reset cache if environment.yml has not changed
         CACHE_NUMBER: 0
       with:
-        path: ~/conda_pkgs_dir
+        path: /usr/local/miniconda/envs/scipy-dev
         key:
           ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('environment.yml') }}
+      id: envcache
```

Comment:
Also, we can use 24-hour cache updates. That is, refreshing cache every day which might be useful on scipy main repo since there are many updates daily and also a large number of CI runs. What are your views on this?
```

      - name: Set cache date
        run: echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
        
      - uses: actions/cache@v2
        with:
          path: ${{ matrix.prefix }}
          key: ${{ matrix.label }}-conda-${{ hashFiles('environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
        env:
          CACHE_NUMBER: 0
        id: envcache
```